java/client/src/org/openqa/selenium/support/ui/Select.java

```diff
@@ -283,15 +282,31 @@ public void deselectByVisibleText(String text) {
     List<WebElement> options = element.findElements(By.xpath(
         ".//option[normalize-space(.) = " + Quotes.escape(text) + "]"));
 
+    boolean matched = false;
     for (WebElement option : options) {
-      if (option.isSelected()) {
-        option.click();
+      setSelected(option, false);
+      if (!isMultiple()) {
+        return;
       }
+      matched = true;
+    }
+
+    if (!matched) {
+      throw new NoSuchElementException("Cannot locate element with text: " + text);
     }
   }
 
-  private void setSelected(WebElement option) {
-    if (!option.isSelected()) {
+  /**
+   * Select or deselect specified option
+   *
+   * @param option
+   *          The option which state needs to be changed
+   * @param select
+   *          Indicates whether the option needs to be selected (true) or
+   *          deselected (false)
+   */
+  private void setSelected(WebElement option, boolean select) {
+    if ((!option.isSelected() && select) || (option.isSelected() && !select)) {
```

Comment:
Hi @lukeis,

Thank you for your comment. It is indeed. Updated in commit f326050e1d611da34dbce9748de6d627efd46f3f.